test/unit/lib/plugins/aws/utils/credentials.test.js

```diff
@@ -57,16 +60,10 @@ describe('#credentials', () => {
       `aws_secret_access_key = ${profile2.secretAccessKey}`,
     ].join('\n')}\n`;
 
-    return new BbPromise((resolve, reject) => {
-      outputFile(credentialsFilePath, credentialsFileContent, (error) => {
-        if (error) reject(error);
-        else resolve();
-      });
-    }).then(() =>
-      credentials
-        .resolveFileProfiles()
-        .then((resolvedProfiles) => expect(resolvedProfiles).to.deep.equal(profiles))
-    );
+    await expect(outputFile(credentialsFilePath, credentialsFileContent)).to.be.fulfilled;
```

Comment:
Let's not use `to.be.fulfilled` anywhere

We should just `await promise` that's it (if it reject, we will have the test fail with a meaningful error information)

Reply:
sure, what the pros/cons of using `to.be.fulfilled` vs not using it

From the top of my head if we wrap our promise around `expect().to.be.fulfilled` we don't get a thrown exception, but rather a failed assert. 
Nevertheless, both will lead to a failed test case. So this might be just a convention choice, let's agree on one and then apply to other places.

Reply:
> From the top of my head if we wrap our promise around expect().to.be.fulfilled we don't get a thrown exception, but rather a failed assert.

Failed assert is a thrown exception, but in this case it's worse exception (than natural promise rejection), as it most likely won't share details of the promise fail, it'll just state that promise didn't fulfill, and we will be left clueless on the real exception details.

So I personally do not see any advantage of `to.be.fulfilled` as it'll hide important debug information in case of test failure and also it adds extra verbosity to tests, which affects their readability.

Reply:
seem reasonable. Does it mean we might avoid `to.be.fulfilled` from now on unless it is necessary?

Reply:
Yes, I'm not sure if there's a good use case for it. If we use it in some places we may as well refactor that out into plain promise usage.

Reply:
sure, I guess that we might turn it into one ticket so that we can keep track. Turning it into a simple `await` statement would be better since we can again avoid nested code.

A sample usage in our code:
```
return expect(serverless.yamlParser.parse(tmpFilePath)).to.be.fulfilled.then((obj) => {
        expect(obj.foo).to.equal('bar');
      });
```

Reply:
>  Turning it into a simple await statement would be better since we can again avoid nested code.

Definitely, now that we've converted all the promise returning functions to _async_ functions. we can get rid of `expect(promise).to.be.fulfilled.then(followingLogic)` by simply converting it to `await promise; followingLogic()`